RFS/src/testFixtures/java/com/rfs/framework/PreloadedDataContainerOrchestrator.java

```
@@ -0,0 +1,285 @@
+        throws InterruptedException
+    {
+        return new HashHelper(
+                getImageId(dockerClient, baseSourceVersion.imageName, pullIfUnavailable),
+                getImageId(dockerClient, dataLoaderImageName, pullIfUnavailable),
+                generatorContainerArgs).hashCode();
+    }
+
+    private String getImageName() {
+        return PRELOADED_IMAGE_BASE_NAME + baseSourceVersion.prettyName;
+    }
+
+    private static DockerClient createDockerClient() {
+        var config = DefaultDockerClientConfig.createDefaultConfigBuilder().build();
+        var httpClient = new ApacheDockerHttpClient.Builder()
+                .dockerHost(config.getDockerHost())
+                .sslConfig(config.getSSLConfig())
+                .maxConnections(100)
+                .connectionTimeout(Duration.ofSeconds(30))
+                .responseTimeout(Duration.ofSeconds(45))
+                .build();
+
+        return DockerClientImpl.getInstance(config, httpClient);
+    }
+
+    private static Image getExistingImage(DockerClient dockerClient, String imageName, String tag,
+                                          boolean pullIfUnavailable) throws InterruptedException {
+        var fullName = formatFullImageName(imageName, tag);
+        for (var image : dockerClient.listImagesCmd().exec()) {
+            for (String s : image.getRepoTags()) {
+                if (fullName.equals(s)) {
+                    return image;
+                }
+            }
+        }
+        if (pullIfUnavailable) {
+            log.warn("Image not found. Pulling image: " + fullName);
+            dockerClient.pullImageCmd(imageName)
+                    .withTag(tag)
+                    .exec(new PullImageResultCallback())
+                    .awaitCompletion(PULL_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            return getExistingImage(dockerClient, imageName, tag, false);
+        }
+        return null;
```

Seems like we should throw here rather than make the caller responsible for a null check

---

Null is pretty reasonable - see this code
```
if (getExistingImage(dockerClient, imageName, tag, false) == null) {
   makeNewImage...
```

Since I'm acting upon it in a pretty expected path, I'd rather not make it look like an exception.